Core/CwCheat.cpp

```diff
@@ -495,18 +495,18 @@ CheatOperation CWCheatEngine::InterpretNextCwCheat(const CheatCode &cheat, size_
 		addr = GetAddress(line1.part1 & 0x0FFFFFFF);
 		if (i < cheat.lines.size()) {
 			const CheatLine &line2 = cheat.lines[i++];
-			const int count = line1.part1 & 0xFFFF;
+			const int count = line2.part1 & 0xFFFF;
 
 			// Doesn't have enough lines to process.
-			if (i + count > cheat.lines.size())
+			if (i - 1 + count > cheat.lines.size())
```

Comment:
Arg, thanks for catching the line1/line2 mixups...

Hmm, trying to think this through.  I may be totally misunderstanding these pointer commands, though.

Count should be the number of additional lines (outside the first 2) that contain instructions.  So in say, this example:

```
_L 0x6000103C 0x000000FF //write FF
_L 0x00000001 0x00000004 //with offset 4
_L 0x20000020 0x20000020 //deref twice adding 0x20 each time
```

Then as of this line, `count = 1`, `i = 2`, and `cheat.lines.size()` == 3.  If count was 2, it'd be invalid (but maybe this is common and it should just reduce count to the available lines?)

So I think `1 + 2 > 3` makes sense.  Subtracting 1 allows for the loop in ExecuteOp to read memory outside of `cheat.lines` by one entry.

I think the actual operations this would perform are:
```
base = 0x08800000 + 0x103C;
base = read32(base) + 0x20;
base = read32(base) + 0x20;
write8(base + 4, 0xFF);
```

I guess there are probably a lot of cases where this sort of "chained deference" cheat is useful... if I'm understanding it right.

-[Unknown]

Reply:
I tend to just write most of my cheats in assembly, those code types are way too confusing to remember. The only places that list those more complicated codes being in japanese like for example [this site](http://ijiro.daiwa-hotcom.com/data/CWC.html) or maybe better [this one](https://www21.atwiki.jp/cwcwiki/pages/90.html) - doesn't make it any easier to swallow:].

 The last part of the test cheat I made so in short:
````
_C0 Test for copy pointer code type
//prepare pointer
_L 0x2000103C 0x0880103C // prepare a pointer pointing to itself(we'll use an offset)
_L 0x20001058 0x12345678 // prepare a value for copy type test
//test code
_L 0x6000103C 0x00000004 //copy 4 bytes
_L 0x00000002 0x0000001C //from offset 0x1C
_L 0x10000020 0x00000000 //to offset 0x20
//should copy value from 0x0880103C+0x1C to 0x0880103C+0x20
//0880105C 12345678
````
 seems to fail again with that last change now, it was based on working cheat from psp that also works on ppsspp now, but honestly I'm not sure if there is even some standard to those codes or each plugin had it's own buggy variant.

Reply:
Ah, I see it now, the old code had:

```c++
for (int i = 1; i < count; i ++ ) {
```

So it's offset by 1?  I wonder what count=0 would do then.  I'll fix the offset.

-[Unknown]